File: src/main/java/ru/bkmz/tehSistem/util/gui/elements/BuilderElements.java
package ru.bkmz.tehSistem.util.gui.elements;

import javafx.beans.value.ChangeListener;
import javafx.beans.value.ObservableValue;
import javafx.scene.control.TextField;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

public class BuilderElements {
    public static final Logger logger = LogManager.getLogger();

    private static void textProperty(TextField textField) {
        textField.textProperty().addListener(new ChangeListener<String>() {
            @Override
            public void changed(ObservableValue<? extends String> observable, String oldValue, String newValue) {
                if (!newValue.matches("[\\d,.-]")) {
                    textField.setText(newValue.replaceAll("[^\\d,.-]", ""));
                }
            }
        });
    }


    public static void textPropertyTime(TextField textField, int size, int maxNumber) {
        textField.textProperty().addListener(new ChangeListener<String>() {
            @Override
            public void changed(ObservableValue<? extends String> observable, String oldValue, String newValue) {
                if (!newValue.matches("[\\d]")) {
                    newValue = newValue.replaceAll("[^\\d]", "");
                    if (newValue.length() > size) {
                        newValue = newValue.substring(0, size);
                    }
                    try {
                        if (Integer.parseInt(newValue) > maxNumber) {
                            newValue = newValue.substring(0, newValue.length() - 1);
                        }
                    } catch (NumberFormatException nfe) {

                    }

                    textField.setText(newValue);
                }
            }
        });
    }

}

File: src/main/kotlin/ru/bkmz/util/gui/elements/BuilderElements.kt
package ru.bkmz.util.gui.elements

import javafx.scene.control.TextField
import org.apache.logging.log4j.LogManager
val logger = LogManager.getLogger()


    private fun textProperty(textField: TextField) {
        textField.textProperty().addListener { _, _, newValue ->
            if (!newValue.matches("[\\d,.-]".toRegex())) {
                textField.text = newValue.replace("[^\\d,.-]".toRegex(), "")
            }
        }
    }

    fun textPropertyTime(textField: TextField, size: Int, maxNumber: Int) {
        textField.textProperty().addListener { _, _, newValue ->
            var newValueVar = newValue
            if (!newValueVar.matches("[\\d]".toRegex())) {
                newValueVar = newValueVar.replace("[^\\d]".toRegex(), "")
                if (newValueVar.length > size) {
                    newValueVar = newValueVar.substring(0, size)
                }
                try {
                    if (newValueVar.toInt() > maxNumber) {
                        newValueVar = newValueVar.substring(0, newValueVar.length - 1)
                    }
                } catch (nfe: NumberFormatException) {
                }
                textField.text = newValueVar
            }
        }
    }



